apps/operate/components/Contracts/hooks.ts

```diff
@@ -0,0 +1,145 @@
+  '0x000000000000000000000000389b46c259631acd6a69bde8b6cee218230bae8c': 'quickstart',
+  '0x0000000000000000000000005344b7dd311e5d3dddd46a4f71481bd7b05aaa3e': 'quickstart',
+};
+
+const getApy = (rewardsPerSecond: bigint, minStakingDeposit: bigint) => {
+  const rewardsPerYear = rewardsPerSecond * BigInt(ONE_YEAR);
+  const apy = (rewardsPerYear * BigInt(100)) / minStakingDeposit;
+  return (Number(apy) / 100).toFixed(2);
+};
+
+const getStakeRequired = (minStakingDeposit: bigint, numAgentInstances: bigint) => {
+  return formatEther(minStakingDeposit + minStakingDeposit * numAgentInstances);
+};
+
+const useContractDetails = (
+  nominees: { account: Address; chainId: number }[],
+  functionName: string,
+) => {
+  const contracts = nominees.map((nominee) => ({
+    address: getAddressFromBytes32(nominee.account),
+    abi: STAKING_TOKEN.abi as Abi,
+    chainId: Number(nominee.chainId),
+    functionName: functionName,
+  }));
+
+  const { data, isFetching } = useReadContracts({
+    contracts,
+    query: {
+      enabled: nominees.length > 0,
+      select: (data) => {
+        return data.map((item) => (item.status === 'success' ? item.result : null));
+      },
+    },
+  });
+
+  return { data, isFetching };
+};
+
+export const useStakingContractsList = () => {
+  // Get nominees list
+  const { data: nomineesData, isFetching: isNomineesLoading } = useNominees();
+  const nominees = (nomineesData || []).filter(
+    (nominee) => nominee.account !== getBytes32FromAddress(RETAINER_ADDRESS),
+  );
+
+  // Get contracts metadata
+  const { data: metadata, isLoading: isMetadataLoading } = useNomineesMetadata(nominees);
+  // Get maxNumServices
+  const { data: maxNumServicesList, isFetching: isMaxNumServicesLoading } = useContractDetails(
+    nominees,
+    'maxNumServices',
+  );
+  // Get serviceIds
+  const { data: serviceIdsList, isFetching: isServiceIdsLoading } = useContractDetails(
+    nominees,
+    'getServiceIds',
+  );
+  // Get rewardsPerSecond
+  const { data: rewardsPerSecondList, isFetching: isRewardsPerSecondLoading } = useContractDetails(
+    nominees,
+    'rewardsPerSecond',
+  );
+  // Get minStakingDeposit
+  const { data: minStakingDepositList, isFetching: isMinStakingDepositLoading } =
+    useContractDetails(nominees, 'minStakingDeposit');
+  // Get numAgentInstances
+  const { data: numAgentInstancesList, isFetching: isNumAgentInstancesLoading } =
+    useContractDetails(nominees, 'numAgentInstances');
+
+  /**
+   * Sets staking contracts list to the store
+   **/
+  const contracts = useMemo(() => {
+    if (
+      // Check if all data is loaded
+      !!nominees &&
+      !!metadata &&
+      !!maxNumServicesList &&
+      !!serviceIdsList &&
+      !!rewardsPerSecondList &&
+      !!minStakingDepositList &&
+      !!numAgentInstancesList
+    ) {
+      return nominees.map((item, index) => {
+        const maxSlots = Number(maxNumServicesList[index]);
```

Comment:
@kupermind please feel free to only review this file, or just simply confirm if the values are correct on the screen
<img width="1225" alt="image" src="https://github.com/user-attachments/assets/4167ce7a-0cc2-4473-922f-ec795384ef17">

Reply:
This must be multiplied by 100, and divided by the sum of (1 + number of agent instances)